### PR TITLE
Adds support for deploying Bicep templates

### PR DESCRIPTION
**Pull request by JamesDawson**
Updated Cmdlets
* `Invoke-ArmTemplateDeployment` - The `ArmTemplatePath` parameter can now reference an [Azure Bicep](https://github.com/Azure/bicep/) file, also adds the `DeploymentScope` parameter to support targetting different ARM deployment scopes

### CODE REVIEW

**Comment by JamesDawson** [comment]
Waiting behind a `corvus.deployment` PR

File: module/functions/azure/_DeployArmArtifacts.ps1 [hunk 1]
```diff
@@ -0,0 +1,114 @@
+# <copyright file="_DeployArmArtifacts.ps1" company="Endjin Limited">
+# Copyright (c) Endjin Limited. All rights reserved.
+# </copyright>
+
+<#
+.SYNOPSIS
+Handles uploading linked ARM templates and other artifacts to a staging sstorage account.
```
Comment (MikeLarah) [inline module/functions/azure/_DeployArmArtifacts.ps1:7]: typo in storage

File: module/functions/azure/_DeployArmArtifacts.ps1 [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+
+.DESCRIPTION
+Handles uploading linked ARM templates and other artifacts to a staging sstorage account.
```
Comment (MikeLarah) [inline module/functions/azure/_DeployArmArtifacts.ps1:10]: typo in storage

File: module/functions/azure/Invoke-ArmTemplateDeployment.ps1 [hunk 3]
```diff
@@ -58,15 +61,17 @@ function Invoke-ArmTemplateDeployment
...
-        [Parameter(Mandatory=$true)]
-        [string] $ResourceGroupName,
-
         [Parameter(Mandatory=$true)]
         [string] $Location,
 
         [Parameter(Mandatory=$true)]
         [string] $ArmTemplatePath,
 
+        [ValidateSet("ResourceGroup","Subscription","Tenant")]
```
Comment (MikeLarah) [inline module/functions/azure/Invoke-ArmTemplateDeployment.ps1:70]: Any reason not to include `ManagementGroup`?

Comment (JamesDawson) [reply module/functions/azure/Invoke-ArmTemplateDeployment.ps1:70]: Nope, an oversight - thanks!

**Review by MikeLarah — APPROVED** [review]

**Pull request merged**